vcs/git/repo.go

```diff
@@ -235,11 +233,19 @@ func (r *Repository) Commits(opt vcs.CommitsOptions) ([]*vcs.Commit, uint, error
 			commits = append(commits, r.makeCommit(c))
 		}
 		total++
-		return total < uint(MaxCommits)
+		// If we want total, keep going until the end.
+		if opt.NoTotal == false {
```

Comment:
I find the double negation harder to read "not no-total" than the longer form "no total is false", but might be just me. I'll change it to short form.